test/integration/test_mail.py

```diff
@@ -0,0 +1,178 @@
+from autosubmit.platforms.platform import Platform
+
+
+@pytest.fixture(scope="module")
+def fake_smtp_server() -> Tuple[int, str]:
+    """Start fake SMTP server container.
+    :return: A tuple with the SMTP port, and the SMTP test server API base URL """
+    smtp_port, api_port = sample(range(3500, 4000), 2)
+    with DockerContainer(image="mailhog/mailhog", remove=True) \
+            .with_bind_ports(1025, smtp_port) \
+            .with_bind_ports(8025, api_port) as container:
+        wait_for_logs(container, 'Serving under')
+        api_base = f"http://127.0.0.1:{api_port}"
+        yield smtp_port, api_base
+        requests.delete(f"{api_base}/api/v2/messages")
+
+
+@pytest.fixture
+def mail_notifier(fake_smtp_server, tmp_path):
+    smtp_port, _ = fake_smtp_server
+
+    def expid_aslog_dir(expid):
+        exp_dir = tmp_path / "aslog" / expid
+        exp_dir.mkdir(parents=True)
+        (exp_dir / "dummy_run.log").write_text("Log entry: simulation started.")
+        return exp_dir
+
+    config = type('Config', (), {
+        'MAIL_FROM': 'notifier@localhost',
+        'SMTP_SERVER': f'127.0.0.1:{smtp_port}',
+        'expid_aslog_dir': staticmethod(expid_aslog_dir),
+    })()
+    return MailNotifier(config)
+
+
+@pytest.fixture
+def mock_platform() -> Platform:
+    mock = Mock(spec=Platform)
+    mock.host = 'localhost'
+    mock.name = 'fake-local'
+    return mock
+
+
+def _check_metadata(
+        emails: list[dict],
+        expected_subject: str,
+        expid: str,
+        sender: str,
+        recipients: list[str]) -> None:
+    subject = [email["Content"]["Headers"]["Subject"][0] for email in emails]
+    assert (expected_subject in s for s in subject)
+    body = [email["Content"]["Body"] for email in emails]
+    assert (expid in b for b in body)
+
+    for email in emails:
+        assert sender in email["Raw"]["From"]
+        for recipient in recipients:
+            assert recipient in email["Raw"]["To"]
+
+
+@pytest.mark.docker
+def test_notify_status_change(mail_notifier, fake_smtp_server):
+    _, api_base = fake_smtp_server
+    expid = 'a000'
+    job_name = 'SIM'
+    to_email = ['test@example.com']
+    requests.delete(f"{api_base}/api/v1/messages")
+
+    mail_notifier.notify_status_change(
+        expid, job_name,
+        Status.VALUE_TO_KEY[Status.RUNNING],  # previous status
+        Status.VALUE_TO_KEY[Status.FAILED],  # new status
+        to_email
+    )
+
+    resp = requests.get(f"{api_base}/api/v2/messages")
+    assert resp.json()["count"] == 1
+    emails = resp.json()["items"]
+    _check_metadata(emails, "status has changed to FAILED",
+                   expid, 'notifier@localhost', to_email)
+
+
+@pytest.mark.docker
+def test_experiment_status(mail_notifier, fake_smtp_server, mock_platform):
+    _, api_base = fake_smtp_server
+    expid = 'a000'
+    to_email = ['test@example.com']
+    requests.delete(f"{api_base}/api/v1/messages")
+
+    platform = mock_platform()
```

Comment:
I think `mock_platform` is used only in this test, but it doesn't hurt to have that as a fixture.

However, I got confused when the IDE alerted for this line. Then, after staring at it for some seconds (haven't had :coffee: yet), I realized it's complaining because there's a `mock_platform` given via fixture/argument of the function, but I think you are calling the function here, instead of getting the `Platform` object injected into the function as fixture?

Surprised it's working fine :dizzy: It should work if you use the `mock_platform` that you passed in the function (just drop the `()`)